internal/listener/listener.go

```diff
@@ -220,3 +221,32 @@ func (l *Listener) DumpIncidents(w http.ResponseWriter, r *http.Request) {
 	enc.SetIndent("", "  ")
 	_ = enc.Encode(encodedIncidents)
 }
+
+func (l *Listener) DumpSchedules(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		_, _ = fmt.Fprintln(w, "GET required")
+		return
+	}
+
+	if !l.checkDebugPassword(w, r) {
+		return
+	}
+
+	l.runtimeConfig.RLock()
+	defer l.runtimeConfig.RUnlock()
+
+	for _, schedule := range l.runtimeConfig.Schedules {
+		fmt.Fprintf(w, "[id=%d] %q:\n", schedule.ID, schedule.Name)
+
+		// Iterate in 30 minute steps as this is the granularity Icinga Notifications Web allows in the configuration.
+		// Truncation to seconds happens only for a more readable output.
+		step := 30 * time.Minute
+		start := time.Now().Truncate(time.Second).Add(step)
```

Comment:
Oh, that `Add()` is indeed a left-over from an earlier version that truncated to 30 minute so that it always gives exactly xx:00 and xx:30 timestamps but that was confusing with how Icinga Notifications Web sets `actual_handoff` (assume its 17:24, you update an entry, then it would have evaluated it for 17:00 and ignore the entry because that's before what was set as `actual_handoff`). That should no longer be an issue when truncating to just seconds though.